substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1255,14 +1257,41 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
 
         int exitStatus = 1;
         try {
+            Path argsFile = null;
+            if (config.useJavaModules()) {
+                /* For Java > 8 we use an argument file to pass the options to the builder */
+                argsFile = Files.createTempFile("native-image", "args");
+                argsFile.toFile().deleteOnExit();
+                Files.write(argsFile, (Iterable<String>) command.stream().skip(1).map(NativeImage::quoteFileArg)::iterator);
+                List<String> atCommand = new ArrayList<>();
+                atCommand.add(command.get(0));
+                atCommand.add("@" + argsFile);
+                command = atCommand;
+            }
+            ProcessBuilder pb = new ProcessBuilder();
+            pb.command(command);
             Process p = pb.inheritIO().start();
             exitStatus = p.waitFor();
+            if (exitStatus != 0 && isVerbose() && argsFile != null) {
+                /* Create a copy of the argument file for failure analysis */
+                Path tmpDir = Paths.get(System.getProperty("java.io.tmpdir"));
+                Path argsFileCopy = tmpDir.resolve("native-image-failure.args");
+                Files.copy(argsFile, argsFileCopy, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
How is the user supposed to find this copy? The location of the temp dir is not obvious on all OSes. For example, on my Mac it's `/var/folders/wr/4yqmgc5978z05qsv8j4sxjlr0000gn/T/` which is pretty hard to guess ;-) I would suggest putting the copy in the current working directory. In any case, you should issue a println saying where it is.

Reply:
How about using https://github.com/oracle/graal/blob/146a5548cf95c4803c10103f1e67a0e59dd93ce7/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java#L695 ?

Reply:
Implemented @dougxc 's suggestion. But personally I'd just a) not delete the temp file and b) print its location (only when `isVerbose()` is true). It's the simplest implementation, it uses temp files in a location meant for temp files and it's still easy to find (once we print the path).

Reply:
I still prefer the option of retaining the original temp file so that the command line can simply be copied and pasted for reproduction. You could achieve this with a "conditional" shutdown hook:
```
        Path argsFile = Files.createTempFile("native-image", "args");
        Path[] argsFileBox = {argsFile};
        Runtime.getRuntime().addShutdownHook(new Thread() {
            @Override
            public void run() {
                if (argsFileBox[0] != null) {
                    argsFileBox[0].toFile().delete();
                }
            }
        });

        ...

        if (exitStatus != 0 && isVerbose() && argsFile != null) {
           argsFileBox[0] = null;
        }
```

Reply:
Like this?

Reply:
Yes.